source/MaterialXCore/Node.cpp

```
@@ -199,6 +199,16 @@ bool Node::validate(string* message) const
         string matchMessage;
         bool exactMatch = hasExactInputMatch(nodeDef, &matchMessage);
         validateRequire(exactMatch, res, message, "Node interface error: " + matchMessage);
+
+        vector<OutputPtr> nodeDefOutputs = nodeDef->getOutputs();
```

Small notes: 

- `getActiveOutputs()` should be used instead of `getOutputs()` as the former handles definition inheritance. 
- Please use MULTI_OUTPUT_TYPE_STRING constant instead of "multioutput".

Note: this is very similar to `NodeDef::getType()` where you can see usage, though that function should probably throw an exception if there are no outputs and not a string.

---

Great feedback! I'm now using `MULTI_OUTPUT_TYPE_STRING` and I've tried to align the style with could be found in `NodeDef::getType`.